### ISSUE TITLE
Can't test macro defined in cljc file

### ISSUE THREAD

**Issue by aengelberg** [cloverage/cloverage#223]
I have a macro defined in a source file like so:
```clojure
;; core.cljc
(ns instaparse.core)


#?(:clj
   (defmacro defparser
     [name grammar & {:as opts}]
     nil))
```
and a test file depending on it like so:
```clojure
;; abnf_test.clj
(ns instaparse.abnf-test
  (:require
    [instaparse.core :refer [defparser]]))
```
But when I run `lein cloverage` I get this error:
```

WARNING: reader-conditional already refers to: #'clojure.core/reader-conditional in namespace: clojure.tools.reader.impl.utils, being replaced by: #'clojure.tools.reader.impl.utils/reader-conditional
WARNING: tagged-literal already refers to: #'clojure.core/tagged-literal in namespace: clojure.tools.reader.impl.utils, being replaced by: #'clojure.tools.reader.impl.utils/tagged-literal
Loading namespaces:  (instaparse.core)
Test namespaces:  (instaparse.abnf-test)
Loaded  instaparse.core  .
Instrumented namespaces.
java.lang.IllegalAccessError: defparser does not exist, compiling:(instaparse/abnf_test.clj:1:1)
```
If I rename `core.cljc` to `core.clj` and remove the reader conditional, the tests pass.

I've pushed this super minimal repro case (including the project.clj) to the `cloverage-repro` branch of instaparse if you want to take a look:
```
$ git clone -b cloverage-repro git@github.com:Engelberg/instaparse.git
$ cd instaparse
$ lein cloverage
```

**Comment by lvh** [comment]
Heya! Thoroughly appreciate the reproducer. As a workaround: can you just get rid of the reader-macro'd macro, or does the original thing have two separate macro definitions?

**Comment by aengelberg** [comment]
I think I need to keep the reader macro, since what I'm accomplishing there is defining a macro in Clojure that is used by both Clojure and ClojureScript.

Also, if I remove the reader conditional, the bug still occurs. I think the bug is dependent on the macro being defined in a cljc file, not the existence of the `#?(:clj)` reader conditional.

**Comment by lvh** [comment]
Ah: you're using bootstrapped Clojurescript? My understanding is macros in regular Clojurescript are always evaluated in Clojure -- that is, during compilation.

I don't have a hypothesis yet for this bug. I can reproduce and can confirm it persists with the reader macro gone (the latter was surprising to me).

**Comment by aengelberg** [comment]
No, I'm using regular ClojureScript, which is why in Instaparse I wrap all my macros in `#?(:clj)` so the defmacro only gets evaluated on Clojure.
